Config Overhaul

MasterEnderman opened the pull request:
Refactored the old config and separated it into multiple models. Everything is localized and can be edited ingame. If you don't find something alarming, I'm going to merge it with the main branch and start refactoring the registration class. Once that is done, we can finally start on merging in the addons.

MasterEnderman committed:
started on overhauling the config system

MasterEnderman committed:
fixed categories

MasterEnderman committed:
reworked dawnstone anvil config

MasterEnderman committed:
unified lang key scheme

MasterEnderman committed:
fixed a typo

MasterEnderman committed:
updated charger config

MasterEnderman committed:
updated cinder plinth and tank config

MasterEnderman committed:
updated reservoir config

MasterEnderman committed:
updated mini boiler config

MasterEnderman committed:
updated ember bore config

MasterEnderman committed:
updated ember bore config

MasterEnderman committed:
updated tool configs

MasterEnderman committed:
extracted compat methods

MasterEnderman committed:
reworked steam engine config

MasterEnderman committed:
reworked ore gen config

MasterEnderman committed:
finished world gen config

MasterEnderman committed:
finished mob config

MasterEnderman committed:
removed old config registration

MasterEnderman committed:
finished material config

MasterEnderman committed:
updated .gitignore

MasterEnderman committed:
updated .gitignore

MasterEnderman committed:
added config for WIP stuff

MasterEnderman committed:
fixed missing WIP files

MasterEnderman committed:
removed the remains of the old config system

MasterEnderman committed:
switched to new config system

MasterEnderman committed:
updated CHANGELOG.md

Klebestreifen commented on src/main/java/teamroots/embers/config/ConfigTool.java:
```diff
@@ -0,0 +1,113 @@
+        @Config.RequiresMcRestart
+        @Config.Name("Damage Rates")
+        @Config.Comment({
+                "Syntax is 'damagetype:rate'.",
```
* [x] `"Syntax is 'damagetype:rate'."` maybe to `"Syntax is: '<damagetype>:<rate>'"`
* [x] Provide an example
* [x] Provide a source to find all damagetypes

Klebestreifen commented on src/main/java/teamroots/embers/gui/GuiFactory.java:
Explain to me why the GuiFactory is no longer needed. I have never used one.

MasterEnderman replied:
A GuiFactory is used as a way to simple add new GUIs to the game. The only GUI that was added was a custom config screen, which is no longer needed as I now use the native config screens that's implemented by forge.

Klebestreifen commented on src/main/java/teamroots/embers/itemmod/ModifierShiftingScales.java:
```diff
@@ -23,15 +23,17 @@
 import teamroots.embers.api.itemmod.ModifierBase;
+import teamroots.embers.config.ConfigTool;
 import teamroots.embers.util.RenderUtil;
 
```
* [ ] Remove blank lines in imports

Klebestreifen commented on src/main/java/teamroots/embers/itemmod/ModifierShiftingScales.java:
```diff
@@ -211,14 +214,12 @@ public void onDrawScreenPre(RenderGameOverlayEvent.Pre event) {
             for (int i = 0; i < segs; i++) {
-                if (i == segs - 1)
-                    u = ((last + 2) % 3) * 9;
+                if (i == segs - 1) u = ((last + 2) % 3) * 9;
```
Single-line if's are not always more readable

Klebestreifen commented on src/main/java/teamroots/embers/proxy/CommonProxy.java:
```diff
@@ -29,6 +28,7 @@
 import teamroots.embers.tileentity.*;
+import teamroots.embers.util.CompatUtil;
 import teamroots.embers.util.OreTransmutationUtil;
 
```
empty line

Klebestreifen commented on src/main/java/teamroots/embers/proxy/CommonProxy.java:
```diff
@@ -29,6 +28,7 @@
+import teamroots.embers.util.CompatUtil;
 import teamroots.embers.util.OreTransmutationUtil;
 
 import java.awt.*;
```
What is AWT needed for here?

MasterEnderman replied:
```java
	public void spawnFireBlast(World world, double x, double y, double z, Color color, float scale, int lifetime) {
		//NOOP
	}
```

It's needed for the Color class.

Klebestreifen commented on src/main/java/teamroots/embers/research/ResearchCategory.java:
```diff
@@ -5,118 +5,115 @@
+    public String name = "";
+    public double u = 192.0;
+    public double v = 0;
+    public ResourceLocation texture = new ResourceLocation("embers:textures/gui/codex_index.png");
```
```java
new ResourceLocation(Embers.MODID, "textures/gui/codex_index.png")
```
should work and is cleaner

Klebestreifen commented on src/main/java/teamroots/embers/research/ResearchManager.java:
```diff
@@ -25,14 +26,15 @@
-import teamroots.embers.research.subtypes.ResearchFakePage;
+import teamroots.embers.util.CompatUtil;
 import teamroots.embers.util.Vec2i;
 
```
empty line

Klebestreifen commented on src/main/java/teamroots/embers/tileentity/TileEntityCharger.java:
```diff
@@ -192,11 +192,9 @@ public void update() {
-				Embers.proxy.playMachineSound(this, SOUND_PROCESS, SoundManager.COPPER_CHARGER_LOOP, SoundCategory.BLOCKS, true, 1.0f, 1.0f, (float)pos.getX()+0.5f,(float)pos.getY()+0.5f,(float)pos.getZ()+0.5f);
-				break;
-		}
+        if (id == SOUND_PROCESS) {
```
As long as only SOUND_PROCESS is checked, the if is maybe better. Otherwise a switch is really better.

MasterEnderman replied:
I don't intend to add more sounds to the mod. There are things that are way more important for now.

Klebestreifen commented on src/main/java/teamroots/embers/tileentity/TileEntityCinderPlinth.java:
Empty line in imports

Klebestreifen commented on src/main/java/teamroots/embers/tileentity/TileEntityDawnstoneAnvil.java:
```diff
@@ -35,124 +36,122 @@
-	public NBTTagCompound getUpdateTag() {
-		return writeToNBT(new NBTTagCompound());
-	}
+    public TileEntityDawnstoneAnvil() {
```
Can be removed

MasterEnderman replied:
Could you explain to me why this can be removed without causing any issues?

Klebestreifen replied:
If there is no constructor, the super constructor is called automatically.

Klebestreifen commented on src/main/java/teamroots/embers/tileentity/TileEntityEmberBore.java:
```diff
@@ -164,10 +172,10 @@ public void update() {
         lastAngle = angle;
         if (isRunning) {
-            angle += 12.0f * speedMod;
+            angle += (float) (12.0f * speedMod);
```
* [x] `angle += 12.0d * speedMod;`
       `angle` and `speedMod` are doubles so work with them

Klebestreifen commented on src/main/java/teamroots/embers/tileentity/TileEntitySteamEngine.java:
```diff
@@ -39,7 +42,7 @@
 
 public class TileEntitySteamEngine extends TileEntity implements ITileEntityBase, ITickable, ISoundController, IExtraCapabilityInformation {
-    class BurningFuel {
+    static class BurningFuel {
```
why static?

MasterEnderman replied:
IntelliJ suggests it, also I didn't touched it in my config refactor.

Klebestreifen commented on src/main/java/teamroots/embers/util/EmberGenUtil.java:
```diff
@@ -6,8 +6,8 @@
-import teamroots.embers.ConfigManager;
 import teamroots.embers.RegistryManager;
+import teamroots.embers.config.ConfigMaterial;
 
```
empty line

MasterEnderman replied:
that's the default formatting everything not minecraft related gets an empty line before it's imported. No idea how to change this in IntelliJ.

Klebestreifen commented on src/main/java/teamroots/embers/world/WorldGenOres.java:
```diff
@@ -6,71 +6,85 @@
-		}
-	}
+
+    private boolean isOreSpawnEnabled(int dimension) {
```
can be `public static`

Klebestreifen commented on src/main/java/teamroots/embers/world/WorldGenSmallRuin.java:
```diff
@@ -107,34 +109,35 @@ else if (state.getBlock().getMetaFromState(state) == 1){
+        }
+    }
+
+	private static boolean isSmallRuinEnabled(int dimension) {
```
why so private?

Klebestreifen reviewed (changes requested):
* [x] Look for empty lines in the imports. The `import static` block may be one line apart.
* [x] Also check for missing spaces in the parameters and so on.

I have noted a few things that could be changed.

Klebestreifen committed:
Merge branch 'rekindled' into config-rework

MasterEnderman committed:
tweaked ConfigTool to include https://github.com/Ender-Development/Embers-Extended-Life/pull/4#discussion_r1403509809

MasterEnderman committed:
https://github.com/Ender-Development/Embers-Extended-Life/pull/4#discussion_r1403527991

MasterEnderman committed:
https://github.com/Ender-Development/Embers-Extended-Life/pull/4#discussion_r1403550352

MasterEnderman committed:
more tweaks for https://github.com/Ender-Development/Embers-Extended-Life/pull/4

MasterEnderman committed:
totally didn't break everything

MasterEnderman committed:
totally didn't break everything

MasterEnderman committed:
no idea why this got reverted like 3 times <.<

MasterEnderman committed:
removed the "category" tag from the config finals

MasterEnderman committed:
removed unnecessary super constructor

Klebestreifen reviewed (approved):
Fine

MasterEnderman committed:
final commit on the config refactor.
I know that not all config entries are working right now. This is related to the fact the the RegistryManager is a total clusterf**k and I really want to refactor it before doing any more harm to it by squeezing my new configs in. This has to do for now. Once we finish the refactor of the RegistryManager and maybe taking a quick pass at the RecipeRegistry, we're finally able to start integrating the oder Embers addons.

Klebestreifen reviewed (approved):
Okay, we'll work on the RegistryManager first to make the last config options available before we merge.

Klebestreifen committed:
[WIP] start registry refactor

Klebestreifen commented:
#6

MasterEnderman committed:
refactored the block registry
i started to refactor all classes that mention the old registry only thing left for now is the RecipeRegistry.

MasterEnderman committed:
updated BlockRegister

MasterEnderman committed:
added ArmorMaterialRegister

MasterEnderman committed:
moved Entities to its own register

MasterEnderman committed:
moved Fluids to its own register

MasterEnderman committed:
moved ItemModifiers to its own register

MasterEnderman committed:
added MaterialRegister

MasterEnderman committed:
added ToolMaterialRegister

MasterEnderman committed:
extracted Items to its own register

MasterEnderman committed:
Refactor for new RegisterManagement

MasterEnderman committed:
Added vintagefix and loliASM to dev-env

MasterEnderman committed:
moved to cleanroom maven updated build.gradle

MasterEnderman committed:
made Embers independend from Baubles and Mystical Mechanics, more registry refacor and fixed some missing localizations

MasterEnderman committed:
finished registry refactor
I know that some of them are a bit scuffed, and they can probably be improved further, but I would like to start on adding more config options while integrating the first few addons next. I guess things some just need to 'work' for now.

MasterEnderman committed:
Merge pull request #5 from Ender-Development/registry-rework

Merged registry refactor into config refactor branch

MasterEnderman committed:
re-implemented the sledgehammer fix for the melter on my end

MasterEnderman committed:
added config for copper cell capacity

MasterEnderman committed:
extracted category classes

MasterEnderman committed:
added config for the capacity of the
creative ember source

MasterEnderman committed:
added ember funnel config options

MasterEnderman committed:
added emitter config options

MasterEnderman committed:
added receiver config options

MasterEnderman committed:
added ignem reactor config options

MasterEnderman committed:
fixed typo

MasterEnderman committed:
added config to change seed drop

MasterEnderman committed:
small refactor

MasterEnderman committed:
updated dependencies

MasterEnderman committed:
added config options for
- catalytic plug
- ember ejector
- ember conduit

MasterEnderman committed:
added config options for
- beam canon
- crystal cell
- ember activator
- pressure refinery
- mixer centrifuge
- hearth coil

MasterEnderman committed:
added config options for ore transmutation

MasterEnderman committed:
updated CHANGELOG.md and README.md

MasterEnderman committed:
updated rfg and dependencies

MasterEnderman committed:
added new stamps (thanks to Invadermonky)

MasterEnderman committed:
added compat for EIO and metal rods

MasterEnderman committed:
Melter now ignores off-hand items

MasterEnderman committed:
updated CHANGELOG.md and README.md

MasterEnderman committed:
improved EIO and TC config description
